### ISSUE TITLE
space-after-type-colon doesn't recognize parens

### ISSUE THREAD

**Issue by zertosh** [gajus/eslint-plugin-flowtype#71]
This is incorrectly triggering the rule:

```js
type Foo = {
  barType: (string | () => void),
  // ^-- warning because of parens
};
```
